### PR TITLE
Update to Python 3.11

### PR DESCRIPTION
**Pull request by harisang**
Not sure if all changes included here are needed, but it seems all checks pass.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ coinpaprika>=0.1.0
 requests>=2.28.1
 safe-cli==0.7.0
 safe-eth-py==5.5.0
-slackclient==2.9.4
+slackclient
```
Comment (bh2smith) [inline requirements.txt]: Maybe we can pin whichever version it is you wound up getting with this?

You could check with

```sh
pip freeze | grep slack
```

File: .github/workflows/pull-request.yaml [hunk 2]
```diff
@@ -8,13 +8,13 @@ jobs:
...
-      - name: Setup Python 3.10
+      - name: Setup Python 3.11
         uses: actions/setup-python@v3
         with:
-          python-version: '3.10'
+          python-version: '3.11'
       - name: Install Requirements
         run:
-          pip install -r requirements.txt
+          pip3 install -r requirements.txt
```
Comment (bh2smith) [inline .github/workflows/pull-request.yaml]: You shouldn't need to do this...

**Review by bh2smith — APPROVED** [review]
Looks good, but I can't imagine why the pip3 changes should be necessary and think you could probably pin a slack client version.

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.10.8
+FROM python:3.11
 COPY . .
-RUN pip install -r requirements.txt
+RUN  install -r requirements.txt
```
Comment (fhenneke) [inline Dockerfile]:
```suggestion
RUN  pip install -r requirements.txt
```

**Review by fhenneke — APPROVED** [review]
There seems to be a small bug in the `Dockefile`. Otherwise it looks good.

With regards to updating requirement: We should definitely update some of the requirements, most dependencies are quite out of date. Can be done in some other PR though.

**Comment by harisang** [comment]
Thanks for the reviews, addressed the comments! Want to actually trigger a run in the cluster to verify it is working as intended before merging

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.10.8
+FROM python:3.11
 COPY . .
-RUN pip install -r requirements.txt
+RUN  pip install -r requirements.txt
```
Comment (bh2smith) [inline Dockerfile]:
```suggestion
RUN pip install -r requirements.txt
```

File: Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
...
 ENTRYPOINT [ "python3", "-m"]
```
Comment (bh2smith) [inline Dockerfile]: New line here...

**Review by bh2smith — APPROVED** [review]

File: Dockerfile (hunk 5 above)
Comment (fhenneke) [inline Dockerfile]:
```suggestion
ENTRYPOINT [ "python3", "-m"]

```

**Pull request merged**